core/app/models/spree/country.rb

```diff
@@ -13,9 +13,13 @@ class Country < Spree::Base
 
     validates :name, :iso_name, presence: true
 
+    before_save :clear_cache
+
     def self.default
-      country_id = Spree::Config[:default_country_id]
-      country_id.present? ? find(country_id) : find_by!(iso: 'US')
+      Rails.cache.fetch("#{Rails.application.class.parent_name.underscore}_default_country") do
+        country_id = Spree::Config[:default_country_id]
+        country_id.present? ? find(country_id) : find_by!(iso: 'US')
```

Comment:
I've seen lots of stores where the country by iso `US` does not exists. I'd not go with this assumption and instead make `default_contry_id` mandatory.